### PR TITLE
chore: pin gcr.io/distroless/static to a hash value

### PR DESCRIPTION
**Pull request by nilo19**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

chore: pin gcr.io/distroless/static to a hash value

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/azclient/client-gen/go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module sigs.k8s.io/cloud-provider-azure/pkg/azclient/client-gen
 
-go 1.20
+go 1.21
```
Comment (nilo19) [inline pkg/azclient/client-gen/go.mod:3]: This is done automatically by ide.

File: health-probe-proxy/Dockerfile
Comment (feiskyer) [inline health-probe-proxy/Dockerfile:1]: do we have trivy scanner running for images built from this Dockerfile?

Comment (nilo19) [reply health-probe-proxy/Dockerfile:1]: No we don't, I will add one.

Comment (nilo19) [reply health-probe-proxy/Dockerfile:1]: Added, please take a look.

**Review by feiskyer — COMMENTED** [review]
/lgtm

**Pull request merged**